File: tests/parsers/fixtures/sample.java
package com.example.sample;

import java.util.List;
import java.util.ArrayList;

/**
 * Sample Java class for parser testing.
 */
public class DataProcessor {
    private final String config;

    public DataProcessor(String config) {
        this.config = config;
    }

    public List<String> process(List<String> data) {
        List<String> result = new ArrayList<>();
        for (String item : data) {
            result.add(transform(item));
        }
        return result;
    }

    private String transform(String item) {
        return item.trim().toLowerCase();
    }
}

File: tests/parsers/fixtures/sample.kt
package com.example.sample

import java.util.Locale

/**
 * Sample Kotlin class for parser testing.
 */
class DataProcessor(private val config: String) {

    fun process(data: List<String>): List<String> {
        return data.map { transform(it) }
    }

    private fun transform(item: String): String {
        return item.trim().lowercase(Locale.getDefault())
    }
}

fun loadData(path: String): List<String> {
    return java.io.File(path).readLines()
}
